### PR TITLE
Remove extra braces. Wait for the correct future.

### PR DESCRIPTION
**Pull request by esteve**
This fixes two issues in the example.

### CODE REVIEW

**Comment by tfoote** [comment]
+1

**Comment by wjwwood** [comment]
Lgtm

File: rclcpp_examples/src/parameters/set_and_get_parameters_async.cpp [hunk 1]
```diff
@@ -49,8 +49,8 @@ int main(int argc, char ** argv)
   }
 
   // Get a few of the parameters just set.
-  auto parameters = parameters_client->get_parameters({{"foo", "baz"}});
-  if (rclcpp::spin_until_future_complete(node, results) !=
+  auto parameters = parameters_client->get_parameters({"foo", "baz"});
```
Comment (dirk-thomas) [inline rclcpp_examples/src/parameters/set_and_get_parameters_async.cpp]: Is this change necessary to fix anything?

Weren't the extra braces necessary for Windows? We also do this in various other locations: https://github.com/ros2/system_tests/blob/master/test_communication/test/message_fixtures.hpp#L130

Comment (esteve) [inline rclcpp_examples/src/parameters/set_and_get_parameters_async.cpp]: Yes, otherwise the example fails with an invalid iterator range error.
